### PR TITLE
Refractor

### LINKED ISSUE #166: Move resample code to repurpose
Move resample code to repurpose

https://github.com/TUW-GEO/pytesmo/blob/master/pytesmo/grid/resample.py

### PR DESCRIPTION
**Pull request by wpreimes**
Remove resample code from pytesmo, add repurpose as dependency for backwards compatibility. 
After this is merged, we can close #166 #167

### CODE REVIEW

**Comment by wpreimes** [comment]
Regarding deleting the ``io`` module (#168), should we move that code somewhere else?

**Comment by wpreimes** [comment]
@sebhahn @cpaulik any objections? Otherwise I will merge this and create a new release

File: environment.yml [hunk 1]
```diff
@@ -22,6 +22,8 @@ dependencies:
     - pygeobase
     - ascat
     - ismn
+    - repurpose
```
Comment (cpaulik) [inline environment.yml]: repurpose is now double

Comment (cpaulik) [reply environment.yml]: Also add the minimum version number that is necessary. Otherwise older versions will be accepted which will break

Comment (cpaulik) [reply environment.yml]: Just seen it in requirements. So scratch the last comment

**Comment by cpaulik** [comment]
Thanks for taking care of this. I think you should bump the release to major version 2 since this completely removes some modules

**Comment by cpaulik** [comment]
Or maybe at least to 0.8

**Comment by wpreimes** [comment]
> 
> 
> Or maybe at least to 0.8

Thanks for the feedback, I will do that

**Pull request merged**